packages/react-data-grid/src/masks/InteractionMasks.js

```diff
@@ -452,23 +627,10 @@ class InteractionMasks extends React.Component {
             columns={columns}
           />
         )}
-        {!isEditorEnabled && this.isGridSelected() && (
-          <SelectionMask
-            selectedPosition={selectedPosition}
-            rowHeight={rowHeight}
-            columns={columns}
-            isGroupedRow={rowData && rowData.__metaData ? rowData.__metaData.isGroup : false}
-            scrollLeft={this.props.scrollLeft}
-          >
-            {this.dragEnabled() && (
-              <DragHandle
-                onDragStart={this.handleDragStart}
-                onDragEnd={this.handleDragEnd}
-                onDoubleClick={this.onDragHandleDoubleClick}
-              />
-            )}
-          </SelectionMask>
-        )}
+        {selectedRangeIsSingleCell(this.state.selectedRange) ?
+          this.getSingleCellSelectView(rowData) :
```

Comment:
Our preference is to break render method up into smaller components, rather than splitting into smaller render methods. Even if the two functions here wont be reused outside of this component, splitting into smaller components  is a good way to decompose fat components, and allows us to easily identify which props and state the rendered components use. 

It also allows to easily isolate those units for independent testing.

https://medium.com/dailyjs/techniques-for-decomposing-react-components-e8a1081ef5da

Reply:
Did you see my previous comment about pulling out components rather than render methods?

I'll do it if you're sure, but I don't think it's a great idea: the components themselves end up being extremely simple (so there's little value in testing them), and take so many props the JSX ends up looking just as bad, if not worse, than before.

Additionally, one of these render methods returns an array of elements, which isn't possible with a single component in anything but the most recent versions of React.

Reply:
As a rule of thumb I would keep the new components in the same file if they are not going to be reused. I thought with destructuring it would look clean to pick the relevant props, and then pass them in with spread syntax. 

But if you tried this and it didn't look so neat, I'm happy to go with this solution. I'm not sure if anyone is using this with earlier versions of React. Unofficially, only React 16 is supported in v4, but the docs should be explicit about this. 

Anyway, this looks good to me